src/evalfunc.c

```diff
@@ -9819,7 +9829,65 @@ f_search(typval_T *argvars, typval_T *rettv)
 {
     int		flags = 0;
 
-    rettv->vval.v_number = search_cmn(argvars, NULL, &flags);
+    rettv->vval.v_number = search_cmn(argvars, NULL, NULL, &flags);
+}
+
+/*
+ * "searchstr()" function
+ */
+    static void
+f_searchstr(typval_T *argvars, typval_T *rettv)
+{
+    pos_T	match_pos;
+    pos_T	end_pos;
+    linenr_T	start_lnum;
+    linenr_T	end_lnum;
+    colnr_T	start_col;
+    colnr_T	end_col;
+    char_u	buf[BUFSIZ];
+    int		multi_line = 0;
+    int		matchlen = 0;
+    int		flags = 0;
+
+    rettv->vval.v_string = NULL;
+    rettv->v_type = VAR_STRING;
+
+    rettv->vval.v_number = search_cmn(argvars, &match_pos, &end_pos, &flags);
+
+    if (rettv->vval.v_number > 0)
+    {
+    match_pos.col -= 1; // search_cmn increments match start col by 1, adjust it for ml_get_pos.
+    start_lnum = match_pos.lnum;
+    start_col = match_pos.col;
+    end_lnum = end_pos.lnum;
+    end_col = end_pos.col;
+    if (end_lnum == start_lnum)
+    {
+    matchlen = end_col - start_col - 1;
+    }
```

Comment:
The Vim coding style is to not use curly braces for a single statement block.  Remove the curly braces here.